### PR TITLE
Feature/snapshot nytimes data

### PR DESCRIPTION
**Pull request by jeremiq**
This is the first part of the work required to swap over the NYTimes data set that we are using for inferring R(t). After this goes in I will switch over the 'read' side of things inside covid-data-model

### CODE REVIEW

File: scripts/update_nytimes_data.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+        sha = self.get_master_commit_sha()
+        stamp = self._stamp()
+        version_path = self.NYTIMES_DATA_ROOT / "version.txt"
+        with version_path.open("w+") as vf:
+            vf.write(f"{sha}\n")
+            vf.write(f"Updated on {stamp}")
+       
+    def update(self) -> None:
+        _logger.info("Updating NYTimes dataset.")
+        tmp_dir = tempfile.TemporaryDirectory()
```
Comment (TomGoBravo) [inline scripts/update_nytimes_data.py:44]: How about using a context manager?
https://docs.python.org/3/library/tempfile.html#examples
`with tempfile.TemporaryDirectory() as tmpdirname:` ?

File: scripts/update_nytimes_data.py [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+        p = pathlib.Path(tmp_dir.name)
+
+        _logger.info("Copying data files")        
+        self.clone_repo_to_dir(self._NYTIMES_MASTER_URL, str(p))
+        repo_dir = p / "covid-19-data-master"
+        for f in self._DATA_TARGET_FILES:
+            _logger.infog("Copying file %s", f)
```
Comment (TomGoBravo) [inline scripts/update_nytimes_data.py:51]: info*g* ?

File: scripts/update_nytimes_data.py [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    DATA_URL = "https://github.com/nytimes/covid-19-data/raw/master/{CSV_FILENAME}"
+
+    NYTIMES_DATA_ROOT = DATA_ROOT / "cases-nytimes"
+
+    # This is the only one being accessed currently, also available are us-states.csv and us.csv
+    _DATA_TARGET_FILES = [
+        "us-counties.csv"
+    ]
+
+    def get_master_commit_sha(self):
```
Comment (igorkofman) [inline scripts/update_nytimes_data.py:30]: Is it worth grabbing data from the pinned commit at 
https://raw.githubusercontent.com/... sha ...?

File: scripts/update_nytimes_data.py [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+        r = requests.get(self._NYTIMES_MASTER_API_URL)
+        return r.json()['commit']['sha']
+
+    def write_version_file(self):
+        sha = self.get_master_commit_sha()
```
Comment (TomGoBravo) [inline scripts/update_nytimes_data.py:35]: Lets not record the sha unless it is actually used to fetch the data. Currently if we get very unlucky a commit could sneak in between when master.zip is fetched and the mast sha is fetched.

Comment (jeremiq) [reply scripts/update_nytimes_data.py:35]: I'll switch the implementation over to grab a pinned version of the data. I was following the convention in update.py -- but we probably want to fix how that fetch is happening as well.

File: scripts/update_nytimes_data.py (hunk 3 above)
Comment (jeremiq) [reply scripts/update_nytimes_data.py:30]: Good call, I'll fix this  a different PR.

**Pull request merged**